### DISCUSSION TITLE
Use of GDI

### DISCUSSION THREAD

**Discussion by chazste** [PlummersSoftwareLLC/HelloAssembly#23] [Ideas]
The tiny app has exactly one call into GDI, specifically SetBkMode() to set the Paint DC background mode to transparent for the call to DrawText(). Removing this one call removes 48 bytes from Tiny.exe and 9 bytes on the Crinkler version and I would expect it to have an equally outsized impact on the shellcode version of the app. One approach would be to simply set the window's background brush to WHITE_BRUSH. This would make the window client area white instead of some kind of gray. Is there another way to convince DrawText() of the window's background color?

**Comment by davepl** [comment]
It would have been smarter if I’d written it that way the first time, but since now the precedent is set, it needs to customize the background color to be a fair comparison, I think.

You might be able to draw text opaque but with the same background color as the window, but I bet that’s going to need a GDI call anyway!

Good catch, but I think we have to keep the custom color. If there’s a better/smaller way to accomplish the same thing, I’d be all for it!

- Dave

**Comment by chazste** [comment]
I agree. That is why I started this as a discussion as opposed to a bug. I did look SetBkColor is also a GDI call so not a win. I have not thought of anything else yet.
